src/nupic/algorithms/Connections.cpp

```diff
@@ -418,78 +389,35 @@ bool segmentIncreasingOrder(const SegmentOverlap& a, const SegmentOverlap& b)
   return a.segment < b.segment;
 }
 
-void Connections::computeActivity(const vector<Cell>& input,
+void Connections::computeActivity(const vector<CellIdx>& input,
                                   Permanence activePermanenceThreshold,
                                   SynapseIdx activeSynapseThreshold,
                                   Permanence matchingPermanenceThreshold,
                                   SynapseIdx matchingSynapseThreshold,
-                                  vector<SegmentOverlap>& outActiveSegments,
-                                  vector<SegmentOverlap>& outMatchingSegments,
-                                  bool recordIteration)
+                                  vector<SegmentOverlap>& activeSegmentsOut,
+                                  vector<SegmentOverlap>& matchingSegmentsOut)
+  const
 {
-  vector<UInt32> numActiveSynapsesForSegment(nextFlatIdx_, 0);
-  vector<UInt32> numMatchingSynapsesForSegment(nextFlatIdx_, 0);
+  SegmentExcitationTally excitations(*this, activePermanenceThreshold,
```

Comment:
This was specifically what I was avoiding. See the documentation above the class https://github.com/numenta/nupic.core/pull/998/files#diff-dff563d2b4cacf6184c85c53ed96cf75R598 and you can see in `activateDendrites` how it gets used for multiple vectors: https://github.com/numenta/nupic.core/pull/998/files#diff-ed53b022acbf152a8f806e4a1cf84ecfR669 which wouldn't be possible if this took in a vector.